src/vm/instruction.ts

```diff
@@ -2,6 +2,29 @@
 // Some instructions spread over two words when they use an immediate value, so
 // representing the first one with this struct is enougth.
 
+import { Int16, SignedInteger16 } from 'primitives/int';
+import { Uint64, UnsignedInteger } from 'primitives/uint';
+import { Err, Ok, Result, VMError } from 'result-pattern/result';
+
+export const HighBitSetError: VMError = {
```

Comment:
Realized we can do this:
```typescript
const DECODING_ERRORS = {
    HIGH_BIT_SET: 'High bit is not zero',
    INVALID_OP1_SRC: 'Invalid Operand 1 Source',
    INVALID_PC_UPDATE: 'Invalid PC Update',
    INVALID_AP_UPDATE: 'Invalid AP Update',
    INVALID_RES_LOGIC: 'Invalid Result Logic',
    INVALID_OPCODE: 'Invalid Opcode'
} as const;
```

Reply:
And then `return Err({message: `InstructionDecodingError ${DECODING_ERRORS.<whatever>}`)`

Reply:
nice will do